tests/crowdsourcing/tasks/qa_data_collection/test_qa_data_collection.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+"""
+End-to-end testing for the QA data collection crowdsourcing task.
+"""
+
+import json
+import os
+import unittest
+
+
+# Inputs
+AGENT_DISPLAY_IDS = ('QA Agent',)
+AGENT_MESSAGES = [("Who was the first reigning pope to ever visit the Americas?",)]
+FORM_MESSAGES = ("Pope Paul VI",)
+FORM_TASK_DATA = ({},)
+# No info is sent through the 'task_data' field when submitting the form
+
+
+try:
+
+    from mephisto.abstractions.blueprints.parlai_chat.parlai_chat_blueprint import (
+        SharedParlAITaskState,
+        BLUEPRINT_TYPE,
+    )
+
+    from parlai.crowdsourcing.tasks.qa_data_collection.run import TASK_DIRECTORY
+    from parlai.crowdsourcing.tasks.qa_data_collection.util import get_teacher
+    from parlai.crowdsourcing.utils.frontend import build_task
+    from parlai.crowdsourcing.utils.tests import AbstractParlAIChatTest
+
+    class TestQADataCollection(AbstractParlAIChatTest, unittest.TestCase):
```

Comment:
sorry i think i've said this in a previous PR but can you remind me why we can't have a wrapper for e.g. `skipUnlessMephisto` instead of this big try/except?

Reply:
Ah, because that'd only decorate the `TestQADataCollection` class, but here it's the Mephisto-based imports (including `AbstractParlAIChatTest`, which this class inherits from) that have to be checked for `ImportError`s

Reply:
you could maybe wrap the imports in a try/except, set a variable like `MEPHISTO_AVAILABLE`, and then use the standard `unittest.skipUnless(MEPHISTO_AVAILABLE)`. i don't feel particularly strongly either way

Reply:
Yes, that's a good idea - but maybe that'd be better as a separate PR, since I have a couple of PRs open and a clean implementation of this would affect all of them? But I've added this as a TODO on the feature-tracker list

Reply:
sounds good to me!